tenacity/__init__.py

```diff
@@ -251,8 +251,15 @@ def iter(self, result, exc_info, start_time):
             six.raise_from(RetryError(fut), fut.exception())
 
         if self.wait:
-            sleep = self.wait(self.statistics['attempt_number'],
-                              delay_since_first_attempt)
+            # This will allow for passing in the result and handling
+            # the older versions of these functions that do not take
+            # the prior result.
+            try:
+                sleep = self.wait(self.statistics['attempt_number'],
+                                  delay_since_first_attempt, last_result=fut)
+            except TypeError:
```

Comment:
I don't like this capture here.... other option would be using `isinstance` of a new baseclass...

Reply:
did you try to `inspect` the signature?

Reply:
I didn't though that should work.